### PR TITLE
feat: enable lifi cross-account trades

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Add cross-account trading to LifI.

Cross-account LiFi transactions created from this branch:

- https://snowtrace.dev/tx/0x686bc6eb039a38b9c0fd5a5e8d03307a8256cffad042a82fe2128c31a73921d0

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Relates to https://github.com/shapeshift/web/issues/4966

## Risk

Medium. It's a small change, but as always, touching account selection is risky as mistakes can mean lost funds.

## Testing

Confirm that Li.Fi quotes now show up when trading between accounts, and that they and sent and received from the correct accounts when executed.

### Engineering

Sanity check logic.

### Operations

☝️

## Screenshots (if applicable)

Screenshot 2023-12-12 at 11 30 13 am

### CODE REVIEW

File: src/state/helpers.ts [hunk 1]
```diff
@@ -5,11 +5,8 @@ import { assertUnreachable } from '../lib/utils'
 export const isCrossAccountTradeSupported = (swapperName: SwapperName) => {
   switch (swapperName) {
     case SwapperName.Thorchain:
-      return true
-    // NOTE: Before enabling cross-account for LIFI and OneInch - we must pass the sending address
-    // to the swappers up so allowance checks work. They're currently using the receive address
-    // assuming it's the same address as the sending address.
```
Comment (0xApotheosis) [inline src/state/helpers.ts:11]: Removing this comment now, as LiFi is fixed in this PR (the receive address was actually already set correctly), and 1inch in https://github.com/shapeshift/web/pull/5834.

File: src/lib/swapper/swappers/LifiSwapper/getTradeQuote/getTradeQuote.ts [hunk 2]
```diff
@@ -72,9 +72,6 @@ export async function getTradeQuote(
     toChainId: Number(fromChainId(buyAsset.chainId).chainReference),
     fromTokenAddress: getLifiEvmAssetAddress(sellAsset),
     toTokenAddress: getLifiEvmAssetAddress(buyAsset),
-    // HACK: use the receive address as the send address
-    // lifi's exchanges may use this to check allowance on their side
-    // this swapper is not cross-account so this works
     fromAddress: sendAddress,
     toAddress: receiveAddress,
```
Comment (0xApotheosis) [inline src/lib/swapper/swappers/LifiSwapper/getTradeQuote/getTradeQuote.ts:76]: This comment appears stale, as the `toAddress` is already the `receiveAddress`.

**Review by gomesalexandre — APPROVED** [review]
Conceptual stamp, haven't managed to consistently get Li.Fi quotes (or any quote for that matter) because of our current timeout issues for the last , but let's retest this and/or get this ops tested if things are looking better in this regard tonight 🙏🏽

**Review by gomesalexandre — APPROVED** [review]
On @woodenfurniture's recommendation, managed to test (after many many tries, fees estimation are still turbo long and borked) with Avalanche as Gnosis fees estimation is all kinds of slow and borked atm, got lucky with it when testing 1inch cross-account but seems borked now.

### Account 0 -> 1

image
image
image
image

Note, while not strictly related to this PR, found some interesting discrepancy in the underlying tool being used by the Li.Fi routing vs. the one announced (cc @woodenfurniture):

image
image

### Account 1 -> 0

image
image
image
image

**Pull request merged**